code/modules/cards/singlecard.dm

```diff
@@ -0,0 +1,242 @@
+
+	if(istype(item, /obj/item/toy/cards/deck))
+		var/obj/item/toy/cards/deck/dealer_deck = item
+		if(!dealer_deck.wielded) // recycle card into deck (if unwielded)
+			dealer_deck.insert(src)
+			user.balloon_alert_to_viewers("puts card in deck", vision_distance = COMBAT_MESSAGE_RANGE)
+			return
+		card = dealer_deck.draw(user)
+
+	if(istype(item, /obj/item/toy/singlecard))
+		card = item
+
+	if(card) // card + card = combine into cardhand
+		if(flip_card)
+			card.Flip()
+
+		if(istype(item, /obj/item/toy/cards/deck))
+			// only decks cause a balloon alert
+			user.balloon_alert_to_viewers("deals a card", vision_distance = COMBAT_MESSAGE_RANGE)
+
+		var/obj/item/toy/cards/cardhand/new_cardhand = new (drop_location(), list(src, card))
+		new_cardhand.pixel_x = src.pixel_x
+		new_cardhand.pixel_y = src.pixel_y
+
+		if(!isturf(loc)) // make a cardhand in our active hand
+			user.temporarilyRemoveItemFromInventory(src, TRUE)
+			new_cardhand.pickup(user)
+			user.put_in_active_hand(new_cardhand)
+		return
+
+	if(istype(item, /obj/item/toy/cards/cardhand)) // insert into cardhand
+		var/obj/item/toy/cards/cardhand/target_cardhand = item
+		target_cardhand.insert(src)
+		return
+
+	var/can_item_write
+	var/marked_cheating_color
+
+	if(istype(item, /obj/item/pen))
+		var/obj/item/pen/pen = item
+		can_item_write = TRUE
+		marked_cheating_color = (pen.colour == "white" && "invisible") || pen.colour
+
+	if(istype(item, /obj/item/toy/crayon))
+		var/obj/item/toy/crayon/crayon = item
+		can_item_write = TRUE
+		marked_cheating_color = (crayon.crayon_color == "mime" && "invisible") || crayon.crayon_color
+
+	if(can_item_write && !blank) // You cheated not only the game, but yourself
+		marked_color = marked_cheating_color
+		to_chat(user, span_notice("You put a [marked_color] mark in the corner of [src] with the [item]. Cheat to win!"))
+		return
+
+	if(can_item_write)
+		if(!user.is_literate())
+			to_chat(user, span_notice("You scribble illegibly on [src]!"))
+			return
+
+		var/cardtext = stripped_input(user, "What do you wish to write on the card?", "Card Writing", "", 50)
+		if(!cardtext || !user.canUseTopic(src, BE_CLOSE))
+			return
+
+		cardname = cardtext
+		blank = FALSE
+		update_appearance()
+		return
+	return ..()
+
+/obj/item/toy/singlecard/attackby_secondary(obj/item/item, mob/living/user, modifiers)
+	attackby(item, user, modifiers, flip_card=TRUE)
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+/obj/item/toy/singlecard/attack_hand_secondary(mob/living/carbon/human/user, modifiers)
+	attack_self(user)
+	return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
+
+/obj/item/toy/singlecard/attack_self_secondary(mob/living/carbon/human/user, modifiers)
+	attack_self(user)
+
+/obj/item/toy/singlecard/attack_self(mob/living/carbon/human/user)
+	if(!ishuman(user) || !user.canUseTopic(src, BE_CLOSE, NO_DEXTERITY, NO_TK, !iscyborg(user)))
+		return
+
+	Flip()
+	if(isturf(src)) // only display tihs message when flipping in a visible spot like on a table
```

Comment:
what?

Reply:
You can flip cards in backpacks, in pockets, and hands. We only want the balloon alert when we are flipping the card on a table.

Reply:
src is of type /obj/item/toy/singlecard, not /turf. this check will always fail

Reply:
src is not a turf, this check will always fail

Reply:
Fixed with https://github.com/tgstation/tgstation/pull/64200/commits/cc1dacf75ea3ee9254b1d0ac3553e53e5856c62c